### PR TITLE
Fix: Kits - Prevent recreate kit when kit not exists. [ED-2394]

### PR DESCRIPTION
**Pull request by Nevoss**
~Depends on #15004~

### CODE REVIEW

File: assets/dev/js/editor/editor-base.js [hunk 1]
```diff
@@ -1091,6 +1098,19 @@ export default class EditorBase extends Marionette.Application {
 		} );
 	}
 
+	kitNotExistsError() {
+		this.showFatalErrorDialog( {
+			className: 'elementor-preview-loading-error',
+			headerMessage: __( 'Your site doesn’t have a default kit', 'elementor' ),
+			message: __( 'Seems like your kit was deleted, please create new one.' ),
```
Comment (StyleShit) [inline assets/dev/js/editor/editor-base.js:1105]:
```suggestion
			message: __( 'Seems like your kit was deleted, please create new one.', 'elementor' ),
```

Comment (StyleShit) [inline assets/dev/js/editor/editor-base.js:1105]:
```suggestion
			message: __( 'Seems like your kit was deleted, please create new one.', 'elementor' ),
```

File: tests/phpunit/base-class.php [hunk 2]
```diff
@@ -3,6 +3,7 @@
 
 use Elementor\Plugin;
 use Elementor\Core\Wp_Api;
+use Elementor\Core\Kits\Manager;
```
Comment (StyleShit) [inline tests/phpunit/base-class.php:6]:
```suggestion
use Elementor\Core\Kits\Manager;
use Elementor\Core\Wp_Api;
```

**Review by StyleShit — CHANGES_REQUESTED** [review]

**Comment by StyleShit** [comment]
Nice Work :)
I would add some PHPDoc for fun 😛

File: tests/phpunit/base-class.php (hunk 2 above)
Comment (StyleShit) [reply tests/phpunit/base-class.php:6]: A-Z

File: assets/dev/js/editor/editor-base.js (hunk 1 above)
Comment (Nevoss) [reply assets/dev/js/editor/editor-base.js:1105]: Nice Catch!

Comment (Nevoss) [reply assets/dev/js/editor/editor-base.js:1105]: Done

File: tests/phpunit/base-class.php (hunk 2 above)
Comment (Nevoss) [reply tests/phpunit/base-class.php]: Done

**Review by StyleShit — APPROVED** [review]

File: tests/phpunit/elementor/core/utils/test-wp-import.php [hunk 3]
```diff
@@ -5,6 +5,12 @@
 use Elementor\Testing\Elementor_Test_Base;
 
 class Test_WP_Import extends Elementor_Test_Base {
+	public function setUp() {
+		parent::setUp();
+
+		$this->remove_default_kit();
```
Comment (matipojo) [inline tests/phpunit/elementor/core/utils/test-wp-import.php:11]: Explain

File: tests/phpunit/elementor/core/utils/test-wp-exporter.php [hunk 4]
```diff
@@ -15,6 +15,12 @@ class Test_WP_Exporter extends Elementor_Test_Base {
 	 */
 	private $expected_errors_found;
 
+	public function setUp() {
+		parent::setUp();
+
+		$this->remove_default_kit();
```
Comment (matipojo) [inline tests/phpunit/elementor/core/utils/test-wp-exporter.php:21]: Exaplain

**Review by matipojo — CHANGES_REQUESTED** [review]
Please resolve the conflict

File: assets/dev/js/editor/editor-base.js [hunk 5]
```diff
@@ -1091,6 +1098,19 @@ export default class EditorBase extends Marionette.Application {
 		} );
 	}
 
+	kitNotExistsError() {
+		this.showFatalErrorDialog( {
+			className: 'elementor-preview-loading-error',
+			headerMessage: __( 'Your site doesn’t have a default kit', 'elementor' ),
```
[the remark this answers is not in the document]
Comment (Nevoss) [reply assets/dev/js/editor/editor-base.js:1104]: Done

File: assets/dev/js/editor/editor-base.js [hunk 6]
```diff
@@ -1091,6 +1098,19 @@ export default class EditorBase extends Marionette.Application {
...
+			message: __( 'Seems like your kit was deleted, please create new one.', 'elementor' ),
```
[the remark this answers is not in the document]
Comment (Nevoss) [reply assets/dev/js/editor/editor-base.js:1105]: Done

File: tests/phpunit/elementor/core/utils/test-wp-exporter.php (hunk 4 above)
Comment (Nevoss) [reply tests/phpunit/elementor/core/utils/test-wp-exporter.php]: Done

File: tests/phpunit/elementor/core/utils/test-wp-import.php (hunk 3 above)
Comment (Nevoss) [reply tests/phpunit/elementor/core/utils/test-wp-import.php]: Done

**Pull request merged**